manager/controllers/app/plotter_controller_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+package app
+
+import (
+	"context"
+	"time"
+
+	app "github.com/mesh-for-data/mesh-for-data/manager/apis/app/v1alpha1"
+	"github.com/mesh-for-data/mesh-for-data/manager/controllers/utils"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Deploy the plotter and check that the result is what was expected.
+func deployAndCheck(namespace string, shouldSucceed bool) {
+	const timeout = time.Second * 30
+	const interval = time.Millisecond * 100
+
+	// Create the plotter yaml from the hard coded testdata file
+	plotter := &app.Plotter{}
+	Expect(readObjectFromFile("../../testdata/plotter.yaml", plotter)).ToNot(HaveOccurred())
+
+	// Set the namespace we received
+	plotter.SetNamespace(namespace)
+
+	// Create the Plotter
+	Expect(k8sClient.Create(context.Background(), plotter)).Should(Succeed())
+
+	// Don't forget to clean up after tests finish
+	plotterKey := client.ObjectKeyFromObject(plotter)
+	defer func() {
+		plotter := &app.Plotter{ObjectMeta: metav1.ObjectMeta{Namespace: plotterKey.Namespace, Name: plotterKey.Name}}
+		_ = k8sClient.Get(context.Background(), plotterKey, plotter)
+		_ = k8sClient.Delete(context.Background(), plotter)
+	}()
+
+	By("Expecting plotter to be created")
+	Eventually(func() error {
+		return k8sClient.Get(context.Background(), plotterKey, plotter)
+	}, timeout, interval).Should(Succeed())
+
+	// Depending on the test being run, sometimes it should succeed and sometimes it shouldn't.
+	if !shouldSucceed {
+		By("Expecting Plotter to never be ready because reconcile should not be called")
+		Eventually(func() bool {
+			Expect(k8sClient.Get(context.Background(), plotterKey, plotter)).To(Succeed())
+			return plotter.Status.ObservedState.Ready
```

Comment:
Isn't `plotter.Status.ObservedState.Ready` always false by default? So while a reconcile is still running it would be false and this test would succeed. I think a solution could be to mandatorily wait for maybe 1-3 intervals and then also check `plotter.Status.ObservedGeneration`.

Reply:
@froesef In the interval defined in the test the reconcile should have run.  However, I added the check for ObservedGeneration as you suggested just in case.